### ISSUE TITLE
Hyperlinks in table headings don't have formatting

### ISSUE THREAD

**Issue by ajh0912** [squidfunk/mkdocs-material#4327]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

Hyperlinks in the heading of a data tables (including footnotes) don't look like hyperlinks, but do function as a hyperlink.

### Expected behaviour

Hyperlinks in the heading should be formatted blue in this case.
Example with MkDocs:
![image](https://user-images.githubusercontent.com/40744246/188719571-2720f430-c05f-446a-8c20-5537f25efcd4.png)

### Actual behaviour

Hyperlinks (and footnotes) in the heading look like normal text (although are bolded due to being in a heading)
![image](https://user-images.githubusercontent.com/40744246/188719614-f7610741-7ec0-4bb0-bf91-5d3881de4207.png)

### Steps to reproduce

test.md:
```
---
hide:
  - toc
  - navigation
---

# Test

| Method[^1] | Description[^2]                      | [This doesn't look like a link][Example Link 1] |
| ---------- | ------------------------------------ | ----------------------------------------------- |
| `GET`      | :material-check:     Fetch resource  | [This looks like a link][Example Link 1]        |
| `PUT`      | :material-check-all: Update resource |                                                 |
| `DELETE`   | :material-close:     Delete resource |                                                 |

These[^1] footnotes look like links when not used in a heading[^2] of a table

[^1]: Lorem ipsum dolor sit amet, consectetur adipiscing elit.

[^2]:
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla et euismod
    nulla. Curabitur feugiat, tortor non consequat finibus, justo purus auctor
    massa, nec semper lorem quam in massa.

[Example Link 1]: https://example.com/1

```

### Package versions

- Python: `Python 3.9.13`
- MkDocs: `mkdocs, version 1.3.0`
- Material: `Version: 8.4.2+insiders.4.22.0`

### Configuration

```yaml
site_name: Test Docs
markdown_extensions:
  - footnotes
  - pymdownx.emoji:
      emoji_index: !!python/name:materialx.emoji.twemoji
      emoji_generator: !!python/name:materialx.emoji.to_svg
```

### System information

- Operating system: macOS 12.4
- Browser: Firefox 104.0.1, Safari 15.5

**Comment by squidfunk** [comment]
Thanks for reporting. Interesting. So it was actually a deliberate decision to override links in table headers, as we did in https://github.com/squidfunk/mkdocs-material/commit/2135e342a2487738efadb4a2a4eb620f539b7fbb and released in [5.2.1](https://github.com/squidfunk/mkdocs-material/releases/tag/5.2.1), but I agree that it doesn't really make sense to do so. However, digging through the issue tracker, it was introduced because of https://github.com/squidfunk/mkdocs-material/issues/1718, which is when the table header had another color than it has now.

For this reason, it's safe to remove, which I just did in 43ea033e4.

**Comment by ajh0912** [comment]
Brilliant, thanks for the quick change.

Can confirm that hyperlinks and footnotes look like they should with the latest version:
![image](https://user-images.githubusercontent.com/40744246/188728240-bf53957f-cf55-4c7b-b63d-cfdaaefc1230.png)

Thanks @squidfunk!

**Comment by squidfunk** [comment]
Released as part of 8.4.3.
